apps/condo/domains/notification/schema/SyncDeviceService.test.js

```diff
@@ -0,0 +1,419 @@
+// noinspection DuplicatedCode
+/**
+ * Generated by `createservice notification.SyncDeviceService --type mutations`
+ */
+const faker = require('faker')
+const sample = require('lodash/sample')
+
+const { makeClient, makeLoggedInClient } = require('@core/keystone/test.utils')
+
+const { syncDeviceByTestClient } = require('@condo/domains/notification/utils/testSchema')
+const { PUSH_TRANSPORT_TYPE_KEYS } = require('@condo/domains/notification/constants/constants')
+
+const getRandomTokenData = (extraAttrs = {}) => {
+    const pushTransport = sample(PUSH_TRANSPORT_TYPE_KEYS)
+    const data =  {
+        deviceId: faker.datatype.uuid(),
+        pushToken: faker.datatype.uuid(),
+        pushTransport,
+        meta: { pushTransport },
+        ...extraAttrs,
+    }
+
+    return data
+}
+
+/**
+ * Test cases for Service
+ *
+ * 1. anonymous/unauthorized (+)
+ * 1.1. anonymous: deviceId + pushTransport (+)
+ *          - register device + pushTransport  (+)
+ *          - ignore update operation (nothing changed)
+ * 1.2. anonymous: deviceId + pushTransport + pushToken (+)
+ *          - register device & pushToken (+)
+ *          - update pushToken of existing device (+)
+ * 1.3. anonymous: deviceId + pushTransport + pushToken + meta (+)
+ *          - register device with pushToken & meta (+)
+ *          - update pushToken with meta of existing device (+)
+ * 1.4. anonymous: deviceId + pushTransport + meta (+)
+ *          - register device with meta (+)
+ *          - update meta of existing device (either no pushToken or pushToken not changed) (+)
+ *
+ * 2. user
+ * 2.1. user: deviceId + pushTransport (+)
+ *          - register device & connect user (+)
+ *          - connect user to existing device (+)
+ *          - reconnect existing device to different user (+)
+ * 2.2. user: deviceId + pushTransport + pushToken (+)
+ *          - register device with pushToken & connect to user (+)
+ *          - update pushToken of existing device & connect to user (+)
+ *          - update pushToken of existing device & reconnect to different user (+)
+ * 2.3. user: deviceId + pushTransport + pushToken + meta (+)
+ *          - register device with pushToken & meta & connect to user (+)
+ *          - update pushToken/meta of existing device & connect to user (+)
+ *          - update pushToken/meta of existing device & reconnect to different user (+)
+ * 2.4. user: deviceId + pushTransport + meta (+)
+ *          - register device with meta & connect to user (no pushToken yet) (+)
+ *          - update meta of existing device & connect to user (no pushToken yet or not changed) (+)
+ *          - update meta of existing device & reconnect to different user (no pushToken yet or not changed) (+)
+ *
+ * 3. admin - the same with user
+ */
+
+describe('SyncDeviceService', () => {
+    test('anonymous: register deviceId + pushTransport + (no pushToken/meta | pushToken | meta | pushToken + meta)', async () => {
```

Comment:
Better to split tests. It can help to distinguish one problem from another later

Reply:
fixed